lib/api/src/lib.rs

```diff
@@ -28,6 +27,199 @@
     )
 )]
 
+//! This crate contains the Wasmer API. The Wasmer API facilitates the efficient,
+//! sandboxed execution of [WebAssembly (Wasm)][wasm] modules.
+//!
+//! For examples of using the Wasmer API, check out the [wasmer examples][wasmer-examples].
+//!
+//! ---------
+//!
+//! # Table of Contents
+//!
+//! - [Project Layout](#project-layout)
+//!   - [Engines](#engines)
+//!   - [Compilers](#compilers)
+//! - [Features](#features)
+//! - [Wasm Primitives](#wasm-primitives)
+//!   - [Externs](#externs)
+//!     - [Functions](#functions)
+//!     - [Memories](#memories)
+//!     - [Globals](#globals)
+//!     - [Tables](#tables)
+//!
+//! ## Project Layout
+//!
+//! The Wasmer project is divided into a number of crates, below is a dependency
+//! graph with transitive dependencies removed.
+//!
+//! <div>
+//! <img src="https://raw.githubusercontent.com/wasmerio/wasmer/master/docs/deps_dedup.svg" />
+//! </div>
+//!
+//! While this crate is the top level API for Wasmer, we also publish crates built
+//! on top of this API that you may be interested in using, including:
+//!
+//! - [wasmer-cache][] for caching compiled Wasm modules.
+//! - [wasmer-emscripten][] for running Wasm modules compiled to the
+//!   Emscripten ABI.
+//! - [wasmer-wasi][] for running Wasm modules compiled to the WASI ABI.
+//!
+//! --------
+//!
+//! The Wasmer project has two major abstractions:
+//! - [Engines][wasmer-engine]
+//! - [Compilers][wasmer-compiler]
+//!
+//! These two abstractions have multiple options that can be enabled
+//! with features.
+//!
+//! ### Engines
+//!
+//! In the wasmer API, an engine is a system that uses a compiler to make
```

Comment:
should we write "**W**asmer" here?

Reply:
I was hoping for feedback on this point!  I was also unsure about what to do in regards to capitalization...  As-is I only uppercased it when referring to the Wasmer org as a whole and lowercased when referring to `wasmer` (the executable and crate named that)...

It's unclear if `wasmer API` should be capitalized though, even given the self-imposed rules I just mentioned...

Either way, thanks for pointing it out, I'm unsure what we want here

Reply:
I like you rule actually!

I think here, as we are talking about an API, we can keep the lowercase `wasmer`. Wasmer, the org, does not have any API :wink:

Reply:
I was about to say the same comment. Either it's Wasmer (for the org), or `wasmer` (in `<code />`) when referring to the crate. Thoughts?